src/mrsimulator/tests/test_mrsimulator_class.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+import os
+
+from mrsimulator import __version__
+from mrsimulator import Mrsimulator
+from mrsimulator import signal_processing as sp
+from mrsimulator import Simulator
+from mrsimulator import Site
+from mrsimulator import SpinSystem
+from mrsimulator.methods import BlochDecaySpectrum
+
+# mrsimulator save and load test
+
+
+__author__ = "Matthew D. Giammar"
+__email__ = "giammar.7@osu.edu"
+
+
+def setup_sim_and_processor():
+    site = Site(isotope="1H", shielding_symmetric={"zeta": -100, "eta": 0.3})
+    spin_sys = SpinSystem(sites=[site, site], abundance=45)
+
+    method = BlochDecaySpectrum(channels=["1H"])
+    sim = Simulator(spin_systems=[spin_sys, spin_sys], methods=[method, method])
+
+    sim.run(method_index=0)
+
+    processor = sp.SignalProcessor(
+        operations=[
+            sp.IFFT(),
+            sp.apodization.Exponential(FWHM="2500 Hz"),
+            sp.FFT(),
+            sp.Scale(factor=20),
+        ]
+    )
+    processors = [processor] * 2
+
+    application = {
+        "com.github.DeepanshS.mrsimulator": {"foo": "This is some metadata"},
+        "com.github.DeepanshS.mrsimulator-app": {"params": "The JSON string of params"},
+    }
+
+    return sim, processors, application
+
+
+def setup_mrsimulator_obj():
+    sim, processors, application = setup_sim_and_processor()
+
+    return Mrsimulator(
+        simulator=sim,
+        signal_processors=processors,
+        application=application,
+        version="an old version",
```

Comment:
This should raise and error. The version should not be set by the user but auto set by the library using the __version__ variable.

Reply:
Will add test raising an error when a user tries to provide a version number along with making `version` a protected attribute

Reply:
Pytest now covers raising a `ValidationError` if a version is passed